app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1026,8 +1026,6 @@ public void onPositiveButtonClick(Long selection) {
                             (Date) dateField.getTag(),
                             // if the expiry date is being set, set date picker's minDate to the 'valid from' date
                             loyaltyCardField == LoyaltyCardField.expiry ? (Date) validFromField.getTag() : null,
-                            // if the 'valid from' date is being set, set date picker's maxDate to the expiry date
-                            loyaltyCardField == LoyaltyCardField.validFrom ? (Date) expiryField.getTag() : null,
```

Comment:
Not all of this should not have been removed. We still want to use the expiry date as max, if set.

We can just revert this commit as this change it is out of scope; again, sorry for that.

Alternatively, we'd need to keep this code (and the `maxDate` parameter), remove the `getDefaultMaxDateOfDatePicker()` function and only call `.setEnd(maxDate)` on the `CalendarConstraints.Builder` if `maxDate` is not `null` below instead of always calling it but passing `getDefaultMaxDateOfDatePicker()` when it's `null`.